File: Python/1974. Minimum Time to Type Word Using Special Typewriter.py
class Solution:
    def minTimeToType(self, word: str) -> int:
        res, word = -1, 'a' + word

        for i in range(len(word) - 1):
            res += min(abs(ord(word[i]) - ord(word[i + 1])), 26 - abs(ord(word[i]) - ord(word[i + 1])))
        
        return res + len(word)

File: Java/1974. Minimum Time to Type Word Using Special Typewriter.java
class Solution {
    public int minTimeToType(String word) {
        int ans = -1, t;
        word = "a" + word;
        char temp[] = word.toCharArray();

        for(int i = 0; i < temp.length - 1; i++){
            t = Math.abs(temp[i] - temp[i + 1]);
            ans += Math.min(t, 26 - t);
        }

        return ans + temp.length;
    }
}
